netty/src/main/java/io/grpc/netty/NettyServerBuilder.java

```diff
@@ -227,17 +236,57 @@ public NettyServerBuilder maxHeaderListSize(int maxHeaderListSize) {
     return this;
   }
 
+  /**
+   * Sets a custom KEEPALIVE_TIME. An unreasonably small value will be bumped to a reasonable value,
+   * and {@code Long.MAX_VALUE} nano seconds or an unreasonably large value will disable keep alive.
+   *
+   * @since 1.3.0
+   */
+  public NettyServerBuilder keepAliveTime(long keepAliveTime, TimeUnit timeUnit) {
+    checkArgument(keepAliveTime > 0L, "KEEPALIVE_TIME must be positive");
+    keepAliveTimeInNanos = timeUnit.toNanos(keepAliveTime);
+    if (keepAliveTimeInNanos >= AS_LARGE_AS_INFINITE) {
+      // Bump KEEPALIVE_TIME to infinite. This disables keep alive.
+      keepAliveTimeInNanos = Long.MAX_VALUE;
+    }
+    if (keepAliveTimeInNanos < MIN_KEEPALIVE_TIME_NANO) {
+      // Bump KEEPALIVE_TIME.
+      keepAliveTimeInNanos = MIN_KEEPALIVE_TIME_NANO;
+    }
+    return this;
+  }
+
+  /**
+   * Sets a custom KEEPALIVE_TIMEOUT. An unreasonably small value will be bumped to a reasonable
+   * value. KEEPALIVE_TIMEOUT must be much smaller than KEEPALIVE_TIME.
```

Comment:
Why must `KEEPALIVE_TIMEOUT must be much smaller than KEEPALIVE_TIME` and what does "much smaller" mean? I'd probably just get rid of this restriction.